### PR TITLE
[READY] Swarmer QOL and Balance Update

### PR DESCRIPTION
**Pull request by IndieanaJones**
## About The Pull Request

This update seeks to iron out some of the problems revolving around swarmers and the rework which occurred in June.  The following changes were made:

- Swarmers now teleport hulks away with their beam akin to simple mobs.  Before this change, hulks could not be stopped by swarmers outside of spamming walls, but that could never save the beacon.
- Swarmer drones now automatically follow the main swarmer and will stop to chase targets.  The middle-click mechanic was tedious at best.  Rule of thumb for how it works now, when you move, they move towards you.
- Ventcrawling has been removed.  Unfortunately due to the balance of ventcrawling in its current state, having a swarmer eat the ORM before you could even know they spawned just fails to be fun.  Will be watching to see if further buffs will be needed to compensate for this change.
- Swarmers now start with 20 resources upon being spawned.  This lets them set up some basic defenses or build a drone from the start.
- Swarmer event chance has been brought in line with other midround antagonists, making them slightly more common.
- Added swarmers to Dynamic midround spawns.  Its a personal thing of mine that every antag eventually get added to the system, and that includes swarmers.

## Why It's Good For The Game

Fixes some gripes about swarmers which persisted through the rework, and tackles the swarmer-eat-ORM-instantly issue without removing the possibility entirely.  Also gives some nice QOL changes in the form of drones auto-following you and having hulks no longer be unstoppable, along with 20 resources at spawn to quickly get something going.

## Changelog
:cl:
add: Dynamic Mode can now spawn Swarmers
balance: Swarmers have received a software update, changing some of their capabilities.  Nanotrasen reports a slight increase in strange objects in the neighboring sector.
/:cl:

### CODE REVIEW

File: code/modules/events/swarmer.dm [hunk 1]
```diff
@@ -1,10 +1,9 @@
 /datum/round_event_control/spawn_swarmer
 	name = "Spawn Swarmer Beacon"
 	typepath = /datum/round_event/spawn_swarmer
-	weight = 8
+	weight = 10
 	max_occurrences = 1 //Only once okay fam
-	earliest_start = 30 MINUTES
```
Comment (LemonInTheDark) [inline code/modules/events/swarmer.dm:6]: ?

File: code/modules/swarmers/swarmer.dm [hunk 2]
```diff
@@ -81,6 +80,14 @@
...
 		diag_hud.add_to_hud(src)
 
+/mob/living/simple_animal/hostile/swarmer/Move()
+	. = ..()
+	if(!LAZYLEN(dronelist))
+		return
+	for(var/d in dronelist)
+		var/mob/living/simple_animal/hostile/drone = d
+		if(!drone.target && !HAS_TRAIT(drone, TRAIT_IMMOBILIZED) && isturf(drone.loc))
+			step_to(drone, src)
```
Comment (LemonInTheDark) [inline code/modules/swarmers/swarmer.dm:90]: This isn't instant is it?

File: code/modules/swarmers/swarmer.dm [hunk 3]
```diff
@@ -465,7 +460,11 @@
 	. = ..()
 	if(!.)
 		return
-	if(!istype(target, /mob/living/simple_animal) || !istype(firer, /mob/living/simple_animal/hostile/swarmer))
+	if((!istype(target, /mob/living/simple_animal) && !ishuman(target)) || !istype(firer, /mob/living/simple_animal/hostile/swarmer))
```
Comment (LemonInTheDark) [inline code/modules/swarmers/swarmer.dm:463]: do you really need if(it's not a simple animal, and it's not a human) as a check? what does this catch?

File: code/modules/events/swarmer.dm (hunk 1 above)
Comment (IndieanaJones) [reply code/modules/events/swarmer.dm:6]: Normalizing base midround antagonist spawn rates as they get balanced.  I didn't do this for swarmers when I initially reworked them so I decided to do it here.

File: code/modules/swarmers/swarmer.dm (hunk 3 above)
Comment (IndieanaJones) [reply code/modules/swarmers/swarmer.dm:463]: The beam teleporting cyborgs and other carbons.

File: code/modules/swarmers/swarmer.dm (hunk 2 above)
Comment (LemonInTheDark) [reply code/modules/swarmers/swarmer.dm:90]: s not. what the fuck do you thing step means you dumbass

File: code/modules/swarmers/swarmer.dm (hunk 3 above)
Comment (LemonInTheDark) [reply code/modules/swarmers/swarmer.dm:463]: ahhh I see

**Pull request merged**